lib/private/Files/ObjectStore/ObjectStoreStorage.php

```diff
@@ -812,4 +828,31 @@ public function cancelChunkedWrite(string $targetPath, string $writeToken): void
 	public function setPreserveCacheOnDelete(bool $preserve) {
 		$this->preserveCacheItemsOnDelete = $preserve;
 	}
+
+	/**
+	 * @throws InsufficientStorage
+	 */
+	private function assertTotalSizeLimit(?int $additionalSize): void {
+		if ($this->totalSizeLimit === null) {
+			return;
+		}
+
+		// To avoid iterating all objects in the object store, calculate the sum of the cached sizes of the root folders of all object storages.
+		$qb = Server::get(IDBConnection::class)->getQueryBuilder();
+		$result = $qb->select($qb->func()->sum('f.size'))
+			->from('storages', 's')
+			->leftJoin('s', 'filecache', 'f', $qb->expr()->eq('f.storage', 's.numeric_id'))
+			->where($qb->expr()->like('s.id', 'object::%', IQueryBuilder::PARAM_STR))
+			->andWhere($qb->expr()->eq('f.path', $qb->createNamedParameter('')))
```

Comment:
```suggestion
			->andWhere($qb->expr()->eq('f.parent', $qb->createNamedParameter(-1, IQueryBuilder::PARAM_INT)))
```
I guess this would also work and maybe is also better? Not sure if it makes sense to change or not :woman_shrugging: